Cython/Utility/CythonFunction.c

```diff
@@ -812,6 +822,34 @@ static PyObject * __Pyx_CyFunction_Vectorcall_FASTCALL_KEYWORDS(PyObject *func,
 }
 #endif
 
+#if CYTHON_COMPILING_IN_LIMITED_API
+static PyType_Slot __pyx_CyFunctionType_slots[] = {
+    {Py_tp_dealloc, (void *)__Pyx_CyFunction_dealloc},
+    {Py_tp_repr, (void *)__Pyx_CyFunction_repr},
+    {Py_tp_call, (void *)__Pyx_CyFunction_CallAsMethod},
+    {Py_tp_traverse, (void *)__Pyx_CyFunction_traverse},
+    {Py_tp_clear, (void *)__Pyx_CyFunction_clear},
+    {Py_tp_methods, (void *)__pyx_CyFunction_methods},
+    {Py_tp_members, (void *)__pyx_CyFunction_members},
+    {Py_tp_getset, (void *)__pyx_CyFunction_getsets},
+    {Py_tp_descr_get, (void *)__Pyx_PyMethod_New},
+    {0, 0},
+};
+
+static PyType_Spec __pyx_CyFunctionType_spec = {
+    "cython_function_or_method",
+    sizeof(__pyx_CyFunctionObject),
+    0,
+#ifdef Py_TPFLAGS_METHOD_DESCRIPTOR
+    Py_TPFLAGS_METHOD_DESCRIPTOR |
+#endif
+#ifdef _Py_TPFLAGS_HAVE_VECTORCALL
+    _Py_TPFLAGS_HAVE_VECTORCALL |
```

Comment:
How is this supported now? Wouldn't we need to define the function pointer offset somewhere?

Reply:
TL;DR: Yeah you are right, currently these won't work, I will remove them for now until we add support for them in CPython.

Longer explanation:

For static types, CPython uses the `*offset` fields on the type. Up until recently C Extension heap types were not able to handle these fields (since the type is opaque).

However, I built a way to handle these offsets using the existing CPython type system infrastructure. This is the PR if you are interested: https://github.com/python/cpython/pull/16076

Unfortunately, I only added support for `__dictoffset__ ` and `__weaklistoffset__`. But `__vectorcall_offset__` should be a corollary of that change.

Reply:
I'll mark this as resolved but it is not indeed resolved. I just left a TODO since this would require changes to CPython as well. Still, let me know if you have more questions!